src/components/portrait/portrait.component.js

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import tagComponent from '../../utils/helpers/tags';
+import OptionsHelper from '../../utils/helpers/options-helper';
+import PortraitGravatar from './portrait-gravatar.component';
+import PortraitInitials from './portrait-initials.component';
+import StyledPortrait, { StyledCustomImg, StyledIcon } from './portrait.style';
+
+function Portrait(props) {
+  const {
+    alt, darkBackground, gravatar, initials, shape, size, src
+  } = props;
+
+  const showIcon = Boolean(!src && !initials);
+  const showInitials = Boolean(!src && initials);
+  const showGravatar = Boolean(gravatar);
+  const showCustomImg = Boolean(!gravatar && src);
+
+  return (
+    <StyledPortrait
+      showIcon={ showIcon }
+      showInitials={ showInitials }
+      showGravatar={ showGravatar }
+      showCustomImg={ showCustomImg }
+      { ...props }
+      { ...tagComponent('portrait', props) }
+    >
+
+      {showIcon
+        && (
+          <StyledIcon
+            type='individual'
+            size={ size }
+            shape={ shape }
+            darkBackground={ darkBackground }
+          />
+        )
+      }
+
+      {showInitials
+        && (
+          <PortraitInitials
+            size={ size }
+            initials={ initials }
+            darkBackground={ darkBackground }
+            alt={ alt }
+          />
+        )
+      }
+
+      {showGravatar
+        && (
+          <PortraitGravatar
+            gravatarEmail={ gravatar }
+            size={ size }
+            alt={ alt }
+          />
+        )
+      }
+
+      {showCustomImg
+        && (
+          <StyledCustomImg
+            src={ src }
+            alt={ alt }
+            size={ size }
+            data-element='user-image'
+          />
+        )
+      }
+
+    </StyledPortrait>
+  );
+}
+
+Portrait.propTypes = {
+  /** The size of the Portrait. */
+  size: PropTypes.oneOf(OptionsHelper.sizesFull),
+  /** A custom image URL. */
+  src: (props) => {
+    if (!props.gravatar && !props.src && !props.initials) {
+      throw new Error('Portrait requires a prop of "src", "gravatar" or "initials"');
+    } else if (props.gravatar && props.src) {
+      throw new Error('Portrait requires a prop of "src" or "gravatar" but not both');
+    }
```

Comment:
A few points in reply to your comment:

* _"`src` should be type of `String`"_ – I've now added a check for this in 2574103 :thumbsup:
* After discussion with @ianforster-sage and @edleeks87, we agreed that this validation condition actually isn't required – `(!props.gravatar && !props.src && !props.initials)` – It's OK for these 3 props to all be empty. The `Portrait` component will render the :bust_in_silhouette: icon in this situation. So I've removed this validation condition.
* _"you could move that conditional into the `Protrait` component function body"_ – The validation is currently implemented in `propTypes`, which is [only validated in Development builds](https://reactjs.org/docs/typechecking-with-proptypes.html). If this validation logic was moved into the `Portrait` component function body, it would throw validation errors in Production builds. This would be a breaking change, outside the scope of FE-1412. So I believe this validation logic should remain in `propTypes`.